src/logs/mod.rs

```diff
@@ -187,18 +193,38 @@ pub async fn handler(
                         result?;
                     } else {
                         debug!("SQS TEXT EVENT Detected");
-                        process::sqs_logs(message.clone(), coralogix_exporter.clone(), config)
-                            .await?;
+                        process::sqs_logs(
+                            &mctx,
+                            message.clone(),
```

Comment:
Not really part of this PR but I think we wouldn't need to clone here if we didn't use references in lines 109 and 105